### PR TITLE
Added support for determining rackconnect version

### PR DESCRIPTION
**Pull request by jarosser06**

### CODE REVIEW

File: lib/chef/sugar/rackspace/rackconnect.rb [hunk 1]
```diff
@@ -6,16 +6,30 @@ module Rackconnect
       extend self
 
       def rackconnect_v2?(node)
-        return false unless node.key? 'rackspace'
-        if node['rackspace'].key? 'rackconnect'
-          return node['rackspace']['rackconnect']['enabled']
+        return false unless rackconnected?(node)
+        if node['rackspace']['rackconnect']['version'] == 2
```
Comment (martinb3) [inline lib/chef/sugar/rackspace/rackconnect.rb]: Could this just be `return node['rackspace']['rackconnect']['version'] == 2`?

File: lib/chef/sugar/rackspace/rackconnect.rb [hunk 2]
```diff
@@ -6,16 +6,30 @@ module Rackconnect
...
+          return true
         else
           return false
         end
       end
 
       def rackconnected?(node)
-        rackconnect_v2?(node)
+        return false unless node.key? 'rackspace'
+        if node['rackspace'].key? 'rackconnect'
```
Comment (martinb3) [inline lib/chef/sugar/rackspace/rackconnect.rb]: Could this just be `return node['rackspace'].key? 'rackconnect' && node['rackspace']['rackconnect']['enabled']`?

File: lib/chef/sugar/rackspace/rackconnect.rb [hunk 3]
```diff
@@ -6,16 +6,30 @@ module Rackconnect
...
+        if node['rackspace']['rackconnect']['version'] == 2
+          return true
+        else
+          return false
+        end
+      end
+
+      def rackconnect_v3?(node)
+        return false unless rackconnected?(node)
+        if node['rackspace']['rackconnect']['version'] == 3
```
Comment (martinb3) [inline lib/chef/sugar/rackspace/rackconnect.rb]: Could this just be `return node['rackspace']['rackconnect']['version'] == 3`?

Comment (martinb3) [inline lib/chef/sugar/rackspace/rackconnect.rb]: Or perhaps this whole method could be:

```
return rackconnected?(node) && node['rackspace']['rackconnect']['version'] == 3
```

File: lib/chef/sugar/rackspace/rackconnect.rb (hunk 1 above)
Comment (martinb3) [inline lib/chef/sugar/rackspace/rackconnect.rb]: Or perhaps this whole method could be:

```
return rackconnected?(node) && node['rackspace']['rackconnect']['version'] == 2
```

File: lib/chef/sugar/rackspace/rackconnect.rb (hunk 2 above)
Comment (martinb3) [inline lib/chef/sugar/rackspace/rackconnect.rb]: Or even more succinct,

```
def rackconnected?(node)
  return
    node['rackspace'] &&
    node['rackspace']['rackconnect'] &&
    node['rackspace']['rackconnect']['enabled']
end
```

**Comment by martinb3** [comment]
One final thought -- this could be super short if you had `rackconnected?` return a truthy value instead of a true/false; it could return the version of RC.

Something like:

```
def rackconnected?(node)
  return
    node['rackspace'] &&
    node['rackspace']['rackconnect'] &&
    node['rackspace']['rackconnect']['enabled'] &&
    node['rackspace']['rackconnect']['version']
end

def rackconnect_v3?(node)
  return rackconnected?(node) == 3
end

def rackconnect_v2?(node)
  return rackconnected?(node) == 2
end
```

**Comment by jarosser06** [comment]
node['rackspace'] && node['rackspace']['rackconnect'] etc will return nil instead of false

**Comment by martinb3** [comment]
> node['rackspace'] && node['rackspace']['rackconnect'] etc will return nil instead of false

FWIW, `nil` is still falsey in comparisons, so that's okay:

```
$ irb
irb(main):001:0> !!nil
=> false
irb(main):002:0> nil && true
=> nil
irb(main):003:0> nil || true
=> true
```

**Pull request merged**